frontend/packages/dev-console/src/components/topology/D3ForceDirectedRenderer.tsx

```diff
@@ -1073,8 +1073,8 @@ export default class D3ForceDirectedRenderer extends React.Component<
                 _.includes(draggedNodes, nodeId) ? null : this.renderNode(nodeId),
               )}
               {dragNodeId && this.renderDragItems(draggedEdges, draggedNodes)}
-              {this.renderCreateConnection()}
-              {this.renderMoveConnection()}
+              {nodes.length > 1 && this.renderCreateConnection()}
+              {nodes.length > 1 && this.renderMoveConnection()}
```

Comment:
Better to not create the `moveConnection` in the `initializeMoveConnection` function if there are no nodes that are not already connected. You'll also need to bail out of the drag action.

Reply:
@jeff-phillips-18 I added a check for one node in ```initializeMoveConnection``` and ```onNodeHover```. Please take a look again.